src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1415,15 +1415,83 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
 
         case NI_Vector128_ConvertToDouble:
         case NI_Vector256_ConvertToDouble:
+        case NI_Vector512_ConvertToDouble:
+        {
+            assert(sig->numArgs == 1);
+            assert(varTypeIsLong(simdBaseType) || simdBaseType == TYP_FLOAT);
+            if (IsBaselineVector512IsaSupportedOpportunistically())
+            {
+                if (varTypeIsLong(simdBaseType))
+                {
+                    intrinsic = (simdSize == 16) ? NI_AVX512DQ_VL_ConvertToVector128Double
+                                                 : (simdSize == 32) ? NI_AVX512DQ_VL_ConvertToVector256Double
+                                                                    : NI_AVX512DQ_ConvertToVector512Double;
```

Comment:
nit: We typically do something more like:
```
if (simdSize == 64)
{
    intrinsic = NI_AVX512DQ_ConvertToVector512Double;
}
else if (simdSize == 32)
{
    intrinsic = NI_AVX512DQ_VL_ConvertToVector256Double;
}
else
{
    assert(simdSize == 16);
    intrinsic = NI_AVX512DQ_VL_ConvertToVector128Double;
}
```

That way if we have to expand sizes in the future or opt to support this path for `Vector2/3` via `Vector128`, it also can work with `simdSize == 8` and `simdSize == 12`

Reply:
Same goes for the `else` case right below